### ISSUE TITLE
save ip adapter preprocessor result

### ISSUE THREAD

**Issue by godsgo** [Mikubill/sd-webui-controlnet#2283]
Hello, thanks for your big job done.
I have a question can i save preprocessor result of image? Like we can do with pose or canny, and take it back to controlnet.
For example for creating ipdapter wildcards with saving time for generation without preprocessor
preprocessoring of image do 30-40% time of generation for first time, then it takes 1-2%

**Comment by huchenlei** [comment]
I would say it is a valid request. Currently we don't offer download of binary preprocessor results. It is not just IP Adapter, but also for revision control type.

However, the code we currently have assumes the input to be image in many places, so it would be a very large effort to plumb the change through. Does this become a bottleneck for your workflow?

**Comment by godsgo** [comment]
thanks for your answer, I made a style generator telegram bot with image prompts, this is method is fast for develop new style. Fast style creation - slow generation )

i saw in code cpu usage, maybe i mistake, but if we have chose gpu this can help doing job faster...

class IPAdapterModel(torch.nn.Module):
    def __init__(self, state_dict, clip_embeddings_dim, cross_attention_dim, is_plus, sdxl_plus, is_full):
        super().__init__()
        self.device = "cpu"

I dont readed all code just part, i am not specialist, just thoughts out loud. If its hard to emplement, better look on other enchantments. For me all work perfect, i am just ask. thanks in advance...
